tests/specs/lemmas.k

```diff
@@ -325,11 +336,38 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule #range(M [ N := BUF:ByteArray ], L, K) => BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ] ++ #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))  requires K  >Int 0 andBool L <Int N +Int #sizeByteArray(BUF) andBool L >=Int N                          [simplification]
     rule #range(M [ N := BUF:ByteArray ], L, K) => #range(M, L, K)                                                                                                                                                     requires K  >Int 0                                           andBool L >=Int N +Int #sizeByteArray(BUF) [simplification]
 
+
+    rule #range(M [ N := BUF:ByteArray ], N, K ) => 
+      BUF ++ #range( M, N +Int minInt(K, #sizeByteArray(BUF)), K -Int minInt (K, #sizeByteArray(BUF)))  
+      requires K >Int 0 [simplification]
+
+
     rule 0 <=Int lengthBytes ( _ ) => true [smt-lemma, simplification]
 
     rule BA:ByteArray   ==K #buf(32, DATA) => #buf(32, DATA) ==K              BA                                       [simplification, concrete(BA)]
     rule #buf(32, DATA) ==K BA:ByteArray   =>          DATA  ==Int #asInteger(BA) requires #sizeByteArray(BA) <=Int 32 [simplification, concrete(BA)]
 
+    rule #range( M [ N1 := BUF1:ByteArray ] [ N2 := BUF2:ByteArray ], L, K ) =>
+         #range( M                          [ N2 := BUF2           ], L, K )
+      requires N1 +Int #sizeByteArray(BUF1) <=Int L [simplification]
+
+    rule #range (M, N1 +Int N2, K ) => #range( #range( M, N2, #sizeByteArray(M) -Int N2 ), N1, K ) 
+      [simplification, concrete(N2), concrete(M)]
+
+    rule 0 <Int #asWord (BUF ++ #range( M, #sizeByteArray(BUF), LEN -Int #sizeByteArray(BUF) )) => #sizeByteArray(BUF) ==Int LEN
+      requires M ==K #padToWidth(LEN, .ByteArray) [simplification]
+
+    rule maxUInt160 &Int #asWord (BUF ++ #range( M, #sizeByteArray(BUF), LEN -Int #sizeByteArray(BUF) )) => #asWord (BUF ++ #range( M, #sizeByteArray(BUF), LEN -Int #sizeByteArray(BUF) ))
+      requires #asWord(BUF) <=Int maxUInt160 [simplification]
+
```

Comment:
Also not seeing how this one is obviously true. `BUF` being small enough doesn't seem to mean that the entire thing is small enough to fit in a `pow160`.